pkg/controllers/status/cluster_status_controller.go

```diff
@@ -476,8 +475,12 @@ func getAllocatedResource(podList []*corev1.Pod) corev1.ResourceList {
 	for _, pod := range podList {
 		// When the phase of a pod is Succeeded or Failed, kube-scheduler would not consider its resource occupation.
 		if len(pod.Spec.NodeName) != 0 && pod.Status.Phase != corev1.PodSucceeded && pod.Status.Phase != corev1.PodFailed {
-			// TODO(Garrybest): calculate the init container resource when pod is not scheduled
-			podRes := addPodRequestResource(pod)
+			var podRes *util.Resource
+			if pod.Status.Phase == corev1.PodPending {
+				podRes = util.GetPodResourceRequest(pod)
+			} else {
+				podRes = util.GetPodResourceWithoutInitContainers(pod)
+			}
```

Comment:
Does the `kube-scheduler` don't have to reserve resources for `InitContainers` of `running` Pods?
I'm not sure.

Reply:
You're right. Thanks for reminding. I don't find any resource calculation without counting init containers in `kube-scheduler` or `volcano`. And also I find a description in k8s doc -- `Pod level control groups (cgroups) are based on the effective Pod request and limit, the same as the scheduler`. 

https://kubernetes.io/docs/concepts/workloads/pods/init-containers/#resources